File: app/src/main/java/com/cr4sh/nhlauncher/dialogs/ButtonMenuDialog.java
package com.cr4sh.nhlauncher.dialogs;

import android.annotation.SuppressLint;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;
import android.database.sqlite.SQLiteOpenHelper;
import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatDialogFragment;

import com.cr4sh.nhlauncher.DBHandler;
import com.cr4sh.nhlauncher.DialogUtils;
import com.cr4sh.nhlauncher.Item;
import com.cr4sh.nhlauncher.MainActivity;
import com.cr4sh.nhlauncher.MainUtils;
import com.cr4sh.nhlauncher.MyPreferences;
import com.cr4sh.nhlauncher.R;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;


public class ButtonMenuDialog extends AppCompatDialogFragment {

    MainActivity myActivity;

    public ButtonMenuDialog(MainActivity activity) {
        this.myActivity = activity;
    }
    @SuppressLint("SetTextI18n")
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.button_menu_dialog, container, false);

        MainUtils mainUtils = new MainUtils(myActivity);
        MyPreferences myPreferences = new MyPreferences(myActivity);

        LinearLayout bkg = view.findViewById(R.id.custom_theme_dialog_background);
        TextView title = view.findViewById(R.id.dialog_title);
        Button option1 = view.findViewById(R.id.option1);
        Button option2 = view.findViewById(R.id.option2);
        Button option3 = view.findViewById(R.id.option3);
        Button option4 = view.findViewById(R.id.option4);
        Button cancelButton = view.findViewById(R.id.cancel_button);
        DialogUtils dialogUtils = new DialogUtils(requireActivity().getSupportFragmentManager());

        // Apply custom themes
        bkg.setBackgroundColor(Color.parseColor(myPreferences.color20()));
        title.setTextColor(Color.parseColor(myPreferences.color80()));

        option1.setBackgroundColor(Color.parseColor(myPreferences.color50()));
        option1.setTextColor(Color.parseColor(myPreferences.color80()));
        option1.setTextColor(Color.parseColor(myPreferences.color80()));
        option2.setTextColor(Color.parseColor(myPreferences.color80()));
        option2.setBackgroundColor(Color.parseColor(myPreferences.color50()));
        option3.setTextColor(Color.parseColor(myPreferences.color80()));
        option3.setBackgroundColor(Color.parseColor(myPreferences.color50()));
        option4.setTextColor(Color.parseColor(myPreferences.color80()));
        option4.setBackgroundColor(Color.parseColor(myPreferences.color50()));

        cancelButton.setBackgroundColor(Color.parseColor(myPreferences.color80()));
        cancelButton.setTextColor(Color.parseColor(myPreferences.color50()));


        title.setText(myActivity.buttonName.toUpperCase() + " " + requireActivity().getResources().getString(R.string.options).toUpperCase());

        option1.setOnClickListener(view1 -> {
            dialogUtils.openEditableDialog(myActivity.buttonName, myActivity.buttonCmd);
            Objects.requireNonNull(getDialog()).cancel();
        });

        option2.setOnClickListener(view12 -> {
            mainUtils.addFavourite();
            Objects.requireNonNull(getDialog()).cancel();
        });

        option3.setOnClickListener(view1 -> {
            if (!MainActivity.disableMenu) {
                dialogUtils.openNewToolDialog(myActivity.buttonCategory);
            } else {
                Toast.makeText(myActivity, getResources().getString(R.string.get_out), Toast.LENGTH_SHORT).show();
            }
            Objects.requireNonNull(getDialog()).cancel();
        });

        option4.setOnClickListener(view1 -> {
            dialogUtils.openDeleteToolDialog(myActivity.buttonName);
            Objects.requireNonNull(getDialog()).cancel();
        });

        cancelButton.setOnClickListener(view1 -> Objects.requireNonNull(getDialog()).cancel());

        return view;

    }
}

File: app/src/main/java/com/cr4sh/nhlauncher/dialogs/ButtonMenuDialog.kt
package com.cr4sh.nhlauncher.dialogs

import android.annotation.SuppressLint
import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatDialogFragment
import com.cr4sh.nhlauncher.MainActivity
import com.cr4sh.nhlauncher.R
import com.cr4sh.nhlauncher.utils.DialogUtils
import com.cr4sh.nhlauncher.utils.MainUtils
import com.cr4sh.nhlauncher.utils.NHLPreferences
import com.cr4sh.nhlauncher.utils.ToastUtils.showCustomToast
import com.cr4sh.nhlauncher.utils.VibrationUtils.vibrate
import java.util.Locale
import java.util.Objects

class ButtonMenuDialog(private var myActivity: MainActivity) : AppCompatDialogFragment() {
    @SuppressLint("SetTextI18n")
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.button_menu_dialog, container, false)
        val mainUtils = MainUtils(myActivity)
        val nhlPreferences = NHLPreferences(myActivity)
        val bkg = view.findViewById<LinearLayout>(R.id.custom_theme_dialog_background)
        val title = view.findViewById<TextView>(R.id.dialog_title)
        val description = view.findViewById<TextView>(R.id.dialog_description)
        val option1 = view.findViewById<Button>(R.id.option1)
        val option2 = view.findViewById<Button>(R.id.option2)
        val option3 = view.findViewById<Button>(R.id.option3)
        val option4 = view.findViewById<Button>(R.id.option4)
        val cancelButton = view.findViewById<Button>(R.id.cancel_button)
        val dialogUtils = DialogUtils(requireActivity().supportFragmentManager)

        // Apply custom themes
        bkg.setBackgroundColor(Color.parseColor(nhlPreferences.color20()))
        title.setTextColor(Color.parseColor(nhlPreferences.color80()))
        description.setTextColor(Color.parseColor(nhlPreferences.color50()))
        option1.setBackgroundColor(Color.parseColor(nhlPreferences.color50()))
        option1.setTextColor(Color.parseColor(nhlPreferences.color80()))
        option1.setTextColor(Color.parseColor(nhlPreferences.color80()))
        option2.setTextColor(Color.parseColor(nhlPreferences.color80()))
        option2.setBackgroundColor(Color.parseColor(nhlPreferences.color50()))
        option3.setTextColor(Color.parseColor(nhlPreferences.color80()))
        option3.setBackgroundColor(Color.parseColor(nhlPreferences.color50()))
        option4.setTextColor(Color.parseColor(nhlPreferences.color80()))
        option4.setBackgroundColor(Color.parseColor(nhlPreferences.color50()))
        cancelButton.setBackgroundColor(Color.parseColor(nhlPreferences.color80()))
        cancelButton.setTextColor(Color.parseColor(nhlPreferences.color50()))
        title.text = Objects.requireNonNull(myActivity.buttonName).uppercase(Locale.getDefault())
        description.text =
            (Objects.requireNonNull(myActivity.buttonDescription).uppercase(Locale.getDefault())
                ?: option1.setOnClickListener { view1: View? ->
                    vibrate(myActivity, 10)
                    dialogUtils.openEditableDialog(myActivity.buttonName, myActivity.buttonCmd)
                    Objects.requireNonNull(dialog).cancel()
                }) as CharSequence?
        option2.setOnClickListener { view12: View? ->
            vibrate(myActivity, 10)
            mainUtils.addFavourite()
            Objects.requireNonNull(dialog).cancel()
        }
        option3.setOnClickListener {
            vibrate(myActivity, 10)
            if (!MainActivity.disableMenu) {
                dialogUtils.openNewToolDialog(myActivity.buttonCategory)
            } else {
                showCustomToast(requireActivity(), resources.getString(R.string.get_out))
            }
            Objects.requireNonNull(dialog).cancel()
        }
        option4.setOnClickListener {
            vibrate(myActivity, 10)
            dialogUtils.openDeleteToolDialog(myActivity.buttonName)
            Objects.requireNonNull(dialog).cancel()
        }
        cancelButton.setOnClickListener {
            vibrate(myActivity, 10)
            Objects.requireNonNull(dialog).cancel()
        }
        return view
    }
}